web/server/models.py

```diff
@@ -362,3 +372,136 @@ class ArtifactChange(BaseModel):
     path: str
     type: t.Optional[ArtifactType] = None
     file: t.Optional[File] = None
+
+
+class ReportBackfillProgress(BaseModel):
+    completed: int
+    total: int
+    view_name: str
+    start: TimeLike
+    end: t.Optional[TimeLike] = None
+
+
+class ReportTestsResult(BaseModel):
+    message: str
+
+
+class ReportTestDetails(ReportTestsResult):
+    details: str
+
+
+class ReportTestsFailure(ReportTestsResult):
+    total: int
+    failures: int
+    errors: int
+    successful: int
+    dialect: str
+    details: t.List[ReportTestDetails]
+    traceback: str
+
+
+class ReportMeta(BaseModel):
+    status: ReportStatus = ReportStatus.init
+    start_at: int = Field(default_factory=now_timestamp)
+    stop_at: t.Optional[int] = None
+    done: bool = False
+
+    @property
+    def duration(self) -> int | None:
+        return self.stop_at - self.start_at if self.start_at and self.stop_at else None
+
+    def dict(self, *args: t.Any, **kwargs: t.Any) -> t.Dict[str, t.Any]:
+        data = super().dict(*args, **kwargs)
+        data["duration"] = self.duration
+        return data
+
+
+class Stage(BaseModel):
```

Comment:
It's difficult for me to keep track of these models, especially when they're so similarly named. For example, `Stage`, `Report` that is a subclass of Stage, and `ReportStage` that is also a subclass of Stage. Maybe more descriptive names or add some docstrings.